libr/io/p/io_mach.c

```diff
@@ -443,41 +422,29 @@ static ut64 __lseek(RIO *io, RIODesc *fd, ut64 offset, int whence) {
 }
 
 static int __close(RIODesc *fd) {
-	if (!fd) {
-		return false;
-	}
-	RIODescData *iodd = fd->data;
+	RIOMach *riom = (RIOMach*)fd->data;
 	kern_return_t kr;
-	if (!iodd) {
-		return false;
-	}
-	if (iodd->magic != R_MACH_MAGIC) {
+	if (!riom)
 		return false;
-	}
-	task_t task = pid_to_task (iodd->pid);
-	kr = mach_port_deallocate (mach_task_self (), task);
-	if (kr != KERN_SUCCESS) {
+	kr = mach_port_deallocate (mach_task_self (), riom->task);
+	if (kr != KERN_SUCCESS)
 		perror ("__close io_mach");
-	}
 	R_FREE (fd->data);
 	return kr == KERN_SUCCESS;
 }
 
 static int __system(RIO *io, RIODesc *fd, const char *cmd) {
+	RIOMach *riom;
 	if (!io || !fd || !cmd || !fd->data) {
 		return 0;
 	}
-	RIODescData *iodd = fd->data;
-	if (iodd->magic != R_MACH_MAGIC) {
-		return false;
-	}
-	
-	task_t task = pid_to_task (iodd->tid);
+	riom = (RIOMach*)fd->data;
```

Comment:
Accessing private firlds regression

Reply:
this is in the plugin, the plugin may do this